nvflare/fuel/utils/attributes_exportable.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Dict, List, Tuple
+
+from nvflare.fuel.utils.validation_utils import check_object_type
+
+
+class ExportMode:
+    SELF = "SELF"
+    PEER = "PEER"
+
+
+class AttributesExportable(ABC):
+    """Export attributes."""
+
+    @abstractmethod
+    def export(self, export_mode: str) -> Tuple[str, dict]:
+        """Exports attributes.
+
+        Args:
+            export_mode (str): export to peer (`ExportMode.PEER`) or to self (`ExportMode.SELF`).
+
+        Returns:
+            A tuple of (export section name, arguments to be exported)
```

Comment:
I see that you sometimes return class name (for pipe) and sometimes return a section name (e.g. metrics). We should have a consistent definition. We can improve this later.